packages-next/types/src/config/index.ts

```diff
@@ -63,16 +63,28 @@ export type DatabaseCommon = {
 
 export type DatabaseConfig = DatabaseCommon &
   (
-    | {
-        adapter: 'prisma_postgresql';
+    | ((
+        | {
+            /** @deprecated: The `adapter` option is deprecated. Please use `{ provider: 'postgresql' }` */
+            adapter: 'prisma_postgresql';
+            provider?: undefined;
+          }
+        | { adapter?: undefined; provider: 'postgresql' }
```

Comment:
@mitchellhamilton Is there a better way of saying "one or the other, but not both"?

Reply:
Not really